src/RelayProvider.ts

```diff
@@ -0,0 +1,270 @@
+                transactionRelayed: false,
+                reason: revertReason
+            };
+        }
+
+        const transactionRelayed = parsedLogs.find(log => log.name == 'TransactionRelayed');
+
+        if (transactionRelayed) {
+            return {
+                relayRevertedOnRecipient: false,
+                transactionRelayed: true
+            };
+        }
+
+        // Check if it wasn't a deploy call which does not emit TransactionRelayed events but Deployed events
+        const deployedEvent = parsedLogs.find(log => log.name == 'Deployed');
+
+        if (deployedEvent) {
+            return {
+                relayRevertedOnRecipient: false,
+                transactionRelayed: true
+            };
+        }
+
+        log.info(
+            'Neither TransactionRelayed, Deployed, nor TransactionRelayedButRevertedByRecipient events found. This might be a non-enveloping transaction '
+        );
+
+        return {
+            relayRevertedOnRecipient: false,
+            transactionRelayed: false,
+            reason: 'Neither TransactionRelayed, Deployed, nor TransactionRelayedButRevertedByRecipient events found. This might be a non-enveloping transaction'
+        };
+    }
+
+    private async executeRelayTransaction(
+        envelopingRequest: UserDefinedEnvelopingRequest,
+        requestConfig: RequestConfig
+    ): Promise<RelayingResult> {
+        try {
+
+            const transaction = await this.relayClient.relayTransaction(
+                envelopingRequest, requestConfig
+            );
+
+            const { hash } = transaction;
+
+            if(!hash){
+                throw new Error('Transaction has no hash!');
+            }
+            const receipt = await this.getTransactionReceipt(hash);
+
+            const { relayRevertedOnRecipient, reason } = this._getRelayStatus(receipt);
+
+            if (relayRevertedOnRecipient) {
+                throw new Error(
+                        `Transaction Relayed but reverted on recipient - TxHash: ${hash} , Reason: ${reason ?? 'unknown'}`
+                );
+            }
+
+            return {
+                transaction,
+                receipt,
+            };
+        } catch (error) {
+            log.error('Rejected relayTransaction call', error);
+            throw new Error(
+                `Rejected relayTransaction call - Reason: ${(error as Error).message}`
+            );
+        }
+    }
+
+    private async _ethSendTransaction(
+        envelopingRequest: UserDefinedEnvelopingRequest,
+        requestConfig: RequestConfig
+    ): Promise<RelayingResult> {
+        const config = getEnvelopingConfig();
+
+        const { relayData, request } = envelopingRequest;
+
+        let callForwarderValue = relayData.callForwarder;
+        let relayHubValue = request.relayHub;
+        let onlyPreferredRelaysValue = requestConfig.onlyPreferredRelays;
+        let gasToSend = requestConfig.forceGasPrice;
```

Comment:
I think we should use `forceGasLimit` instead

Reply:
got it